plugins/poll/assets/javascripts/components/poll-breakdown-chart.js.es6

```diff
@@ -0,0 +1,175 @@
+import I18n from "I18n";
+import Component from "@ember/component";
+import { htmlSafe } from "@ember/template";
+import discourseComputed from "discourse-common/utils/decorators";
+import { PIE_CHART_TYPE } from "../controllers/poll-ui-builder";
+import { getColors } from "../lib/chart-colors";
+
+// args: options, group, displayMode, highlightedOption, setHighlightedOption
+export default Component.extend({
+  classNames: "poll-breakdown-chart-container",
+  optionToSlice: {},
+  previousHighlightedSliceIndex: null,
+  previousDisplayMode: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    const canvas = this.element.querySelector("canvas");
+    this.set(
+      "chart",
+      new window.Chart(canvas.getContext("2d"), this.chartConfig)
+    );
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    if (this.chart) {
+      this._updateDisplayMode();
+      this._updateHighlight();
+    }
+  },
+
+  willDestroy() {
+    this._super(...arguments);
+
+    if (this.chart) {
+      this.chart.destroy();
+    }
+  },
+
+  @discourseComputed("options")
+  data(options) {
+    return options.mapBy("votes");
+  },
+
+  @discourseComputed("optionColors", "index")
+  colorStyle(optionColors, index) {
+    return htmlSafe(`background: ${optionColors[index]};`);
+  },
+
+  @discourseComputed("data", "displayMode")
+  chartConfig(data, displayMode) {
+    const transformedData = [];
+    let counter = 0;
+
+    this.set("optionToSlice", {});
```

Comment:
I know this is not always easy to do, but generally speaking I find set inside a computed to be risky, can we avoid this ?

Reply:
Yeah, I've renamed that property to `_optionToSlice` and since it isn't a tracked prop, I changed it to just `this._optionToSlice = {}`. It basically only acts as a cache, so another method can use this without recomputing.